src/components/atoms/SelectInput/Select/Select.model.ts

```diff
@@ -0,0 +1,7 @@
+import { ISingleOption } from '../SelectInput.model';
+
+export interface ISelect {
+  handleOnClick: (callback: () => ISingleOption[]) => void | undefined;
```

Comment:
Instead of undefined, we can use optional type

```suggestion
  handleOnClick?: (callback: () => ISingleOption[]) => void;
```